tests/eas/load_tracking.py

```diff
@@ -537,6 +540,406 @@ def test_load_avg_behaviour(self, experiment, tasks):
+                        "ref": "task0"
+                    }
+                }
+            }
+        }
+        """
+
+        return {
+            'migration': {
+                'type' : 'rt-app',
+                'conf' : {
+                    'class' : 'custom',
+                    'json' : json_file,
+                    'prefix' : 'mig_test',
+                },
+            }
+        }
+
+    def _get_phases_names(self, tasks):
+        """
+        Return the phases names of a workload
+        :param task: Tasks description
+        :type task: dict
+        :return a list of the names of the phases
+        """
+        task_name = tasks.keys()[0]
+        return tasks[task_name]['phases'].keys()
+
+    def _get_duration_s(self, tasks, phase):
+        """
+        Compute the duration of a given phase
+        :param task: Tasks description
+        :type task: dict
+
+        :param phase: phase for which the duration is computed
+        :type phase: str
+        :return the duration in second of the phase
+        """
+        task_name = tasks.keys()[0]
+        # Get the loop value for a phase
+        loop = tasks[task_name]['phases'][phase]['loop']
+        # Get the period value for a phase in us
+        period_us = tasks[task_name]['phases'][phase]['timer']['period']
+        return (loop * period_us) / 1e6
+
+    def _get_duty_cycle(self, task, phase):
+        """
+        Compute the duty clycle of a task during a given phase.
+        :param task: task for which the duty clycle is computed
+        :type task: dict
+
+        :param phase: phase for which the duty cycle is computed
+        :type phase: str
+        :return the duty cycle in percent of a task for a given phase
+        """
+        # Get the period value for a phase in us
+        period_us = task['phases'][phase]['timer']['period']
+        # Get the run time value for a phase in us
+        run_time_us = task['phases'][phase]['run']
+        return run_time_us * 100 / period_us
+
+    def _get_cpu(self, task, phase):
+        """
+        Return the cpu assigned to a taks for a given phase.
+        :param task: The task to get the cpu from
+        :type task: dict
+
+        :param phase: The phase to get the cpu from
+        :type phase: str
+        :return a list of cpus assigned to a task for a given phase
+        """
+        return task['phases'][phase]['cpus']
+
+    def _get_cpus(self, tasks, phase):
+        """
+        Return the list of CPUs where the different tasks are executed for a
+        given phase.
+        :param tasks: json description of the tasks
+        :type tasks: dict
+
+        :param phase: phase to get the cpu allocation
+        :return a list of CPUs on which tasks are running for the given phase
+        """
+        cpus = []
+        for task in tasks.iteritems():
+            [cpu] = self._get_cpu(task[1], phase)
+            if cpu not in cpus:
+                cpus.append(cpu)
```

Comment:
What about using a `set` in the list? It should take care of duplicates... and then we can convert to a list at return time...

Reply:
Yes it is better with a set. It is not needed to be converted back to a list.
Done.